python/dolfinx/fem/nonlinearproblem.py

```diff
@@ -0,0 +1,109 @@
+# -*- coding: utf-8 -*-
```

Comment:
Could this be combined with `LinearProblem` in `problem.py`? Is nice to keep the number of files and not have a separate file for every class.